src/Murder.Editor/Data/EditorDataManager.cs

```diff
@@ -257,20 +257,33 @@ private void LoadEditorSettings()
                 _gameProfile = FileHelper.DeserializeAsset<GameProfile>(gameProfilePath)!;
             }
 
-            // Create a game profile, if none was provided from the base game or if the game
-            // provides a different one.
-            // TODO: Is there a better way to verify if the profile match?
-            GameProfile profile = CreateGameProfile();
-            if (_gameProfile is null || _gameProfile.GetType() != profile.GetType())
+            // // Create a game profile, if none was provided from the base game or if the game
+            // // provides an incompatible one.
+            if (!ValidateGameProfileType())
             {
                 GameLogger.Warning($"Didn't find {GameProfileFileName} file. Creating one.");
-
+                GameProfile profile = CreateGameProfile();
                 GameProfile = profile;
                 GameProfile.MakeGuid();
                 SaveAsset(GameProfile);
             }
         }
 
+        private bool ValidateGameProfileType()
+        {
+            if (_gameProfile == null)
+            {
+                return false;
+            }
+            var profileType = _gameProfile.GetType();
+            if (profileType.IsAssignableTo(typeof(GameProfile)))
```

Comment:
so we need this to be the same type (instead of assignable) if a GameProfile was implemented in `IMurderGame` with `IMurderGame.CreateGameProfile`. It's sort of a "workaround" I did to make sure we override the game profile with the custom one.

Reply:
can you let me know if the problem goes away if you implement `IMurderGame.CreateGameProfile`? otherwise we can think of a better way to fix this! (I think I had a TODO there because I wanted to avoid doing a `new()` every time the editor boots, but it's not the end of the world)

Reply:
🤦 yep this isn't a problem when I implemented `IMurderGame.CreateGameProfile`, thank you! I'll close this out.